sympy/physics/units/unitsystem.py

```diff
@@ -26,7 +28,7 @@ class UnitSystem(_QuantityMapper):
 
     _unit_systems = {}  # type: tDict[str, UnitSystem]
 
-    def __init__(self, base_units, units=(), name="", descr="", dimension_system=None):
+    def __init__(self, base_units, units=(), name="", descr="", dimension_system=None, derived_units: tDict[Symbol, Quantity]={}):
```

Comment:
What about mapping `tDict[Dimension, Quantity]` instead of the name of the dimension?

Reply:
That might work, I'm not sure if `Dimension` can be used as a dictionary key.